### PR TITLE
feat: nicer error if transmission session connection fails

### LINKED ISSUE #6: throw a user friendly error when rustmission can't reach transmission

### PR DESCRIPTION
**Pull request by aidanaden**
closes #6

@micielski thoughts on error message? i wrote smth simple but do u think we should also direct the user to the config file path?

### CODE REVIEW

**Comment by micielski** [comment]
> @micielski thoughts on error message?
I think something like this would be better:
```
Connection to transmission failed.
Is "(insert the url here)" specified in "(config path)" correct?
```
You should be able to get config path from rm_config::CONFIG_PATH, and if not, change its visibility to public
and the url from Config

**Comment by aidanaden** [comment]
> > @micielski thoughts on error message?
> > I think something like this would be better:
> 
> ```
> Connection to transmission failed.
> Is "(insert the url here)" specified in "(config path)" correct?
> ```
> 
> You should be able to get config path from rm_config::CONFIG_PATH, and if not, change its visibility to public and the url from Config

updated !

File: rm-main/src/app.rs [hunk 1]
```diff
@@ -45,7 +45,7 @@ impl Ctx {
             }
             Err(_) => {
                 let url = config.connection.url;
-                return Err(Error::msg(format!("Failed to connect to remote transmission client!\nPlease verify connection and authentication to {url}")));
+                return Err(Error::msg(format!("Connection to transmission failed!\nPlease verify connection and credentials to {url}")));
```
Comment (micielski) [inline rm-main/src/app.rs:48]: format!("{e}\nIs the connection info in {} correct?", rm_config::CONFIG_PATH.get())

File: rm-main/src/app.rs [hunk 2]
```diff
@@ -45,7 +45,7 @@ impl Ctx {
             }
             Err(_) => {
```
Comment (micielski) [inline rm-main/src/app.rs:46]: Err(e) => {
in order to not to ignore it as it may contain useful information for the user

**Review by micielski — COMMENTED** [review]
let's not ignore the error

**Pull request merged**